5.0/en/0x14-V6-Cryptography.md

```diff
@@ -2,62 +2,318 @@
+|--|--|--|--|--|
+| <= 80 | 2TDEA   | L = 1024 <br> N = 160  | k = 1024  | f = 160-223 |
+| 112   | 3TDEA\*  | L = 2048 <br> N = 224  | k = 2048  | f = 224-255 |
+| 128   | AES-128 | L = 3072 <br> N = 256  | k = 3072  | f = 256-383 |
+| 192   | AES-192 | L = 7680 <br> N = 384  | k = 7680  | f = 384-511 |
+| 256   | AES-256 | L = 15360 <br> N = 512 | k = 15360 | f = 512+ |
+
+## V6.3 Cipher Algorithms
+
+Cipher algorithms such as AES and CHACHA20 form the backbone of modern cryptographic practice.
+
+|     #     | Description                                                                                                                                                                                                          | L1  | L2  | L3  | CWE |
+| :-------: | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | :-: | :-: | :-: | :-: |
+| **6.2.4** | Verify that insecure block modes (e.g., ECB) and weak padding schemes (e.g., PKCS#1 v1.5) are not used.                                                                                                              |     |  ✓  |  ✓  | 326 |
+| **6.2.5** | Verify that insecure ciphers, including Triple-DES and Blowfish, are not used but secure ciphers and modes** such as AES with GCM are.                                                                                |  ✓  |  ✓  |  ✓  | 326 |
+| **6.2.6** | Verify that nonces, initialization vectors, and other single-use numbers are not used for more than one encryption key/data-element pair. The method of generation must be appropriate for the algorithm being used. |     |     |  ✓  | 326 |
+| **6.2.7** | Verify that encrypted data is authenticated via signatures, authenticated cipher modes, or HMAC to ensure that ciphertext is not altered by an unauthorized party.                                                   |     |     |  ✓  | 326 |
+| **6.2.8** | Verify that any authenticated signatures are operating in encrypt-then-MAC or encrypt-then-hash modes as required. party.                                                   |     |     |  ✓  | 326 |
+
+### Approved Ciphers
+
+Implementations MUST choose from the ciphers in this list, in order of preference:
+
+| Symmetric Key Algorithms | Reference | L1 | L2 | L3 |
+|--|--|--|--|--|
+| AES-256 | [FIPS 197](https://csrc.nist.gov/pubs/fips/197/final) |     |  ✓  |  ✓  |
+| ChaCha20-Poly1305 | [RFC 8439](https://www.rfc-editor.org/info/rfc8439) |     |  ✓  |  ✓  |
+| AES-192 | [FIPS 197](https://csrc.nist.gov/pubs/fips/197/final) |     |  ✓  |  ✓  |
+| AES-128 | [FIPS 197](https://csrc.nist.gov/pubs/fips/197/final) |  ✓  |  ✓  |  ✓  |
+
+Any other cipher options MUST NOT be used.
```

Comment:
AES-* is a cipher while ChaCha20-Poly1305 is an AE. Should this list mention ChaCha20 only here?